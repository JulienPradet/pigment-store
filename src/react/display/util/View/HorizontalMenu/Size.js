import React from 'react'
import {compose, lifecycle, withState, withHandlers} from 'recompose'
import Switch from 'rc-switch'
import './switch.css'
import styles from './horizontalmenu.css'

const SizeInput = compose(
  withState('auto', 'setAuto', (props) => props.value === 'auto'),
  withState('lastValue', 'setLastValue', (props) => props.value === 'auto' ? 0 : props.value),
  lifecycle({
    componentWillReceiveProps (props) {
      if ((props.auto && props.value === 'auto') || (!props.auto && props.value !== 'auto')) {
        return
      }

      props.setAuto(props.value === 'auto')
      if (props.value !== 'auto') {
        props.setLastValue(props.value)
      }
    }
  }),
  withHandlers({
    toggle: ({name, value, onChange, lastValue, setLastValue, auto, setAuto}) => (active) => {
      if (!active) {
        setAuto(true)
        setLastValue(value)
        onChange({[name]: 'auto'})
      } else {
        setAuto(false)
        onChange({[name]: lastValue})
      }
    },
    onChange: ({name, onChange, setLastValue, auto}) => (e) => {
      if (!auto) {
        setLastValue(e.target.value)
        onChange({[name]: e.target.value})
      }
    }
  })
)(({label, name, value, lastValue, toggle, onChange, auto}) => <div className={styles.sizeInput}>
  <div>
    <Switch checked={!auto} onChange={toggle} />
  </div>
  <div>
    {label}
  </div>
  <div>
    <input type='number' value={value === 'auto' ? lastValue : value} onChange={onChange} disabled={auto} />
    px
  </div>
</div>)

const Size = (
  withHandlers({
    onChange: ({size, onChange}) => (newSize) => onChange({size: Object.assign({}, size, newSize)})
  })
)(({size, onChange}) => <div className={styles.size}>
  <SizeInput label='Width' name='width' value={size.width} onChange={onChange} />
  <SizeInput label='Height' name='height' value={size.height} onChange={onChange} />
</div>)

Size.displayName = 'Size'

export default Size
