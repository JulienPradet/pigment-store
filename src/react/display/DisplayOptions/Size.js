import React from 'react'
import {Group, Item} from '../util/View/HorizontalMenu'

const Size = ({size, onChange}) => <Group>
  <Item>
    <input type='text' value={size.width} onChange={(event) => onChange({ width: event.currentTarget.value })} />
    px
  </Item>
  <Item>
    <input type='text' value={size.height} onChange={(event) => onChange({ height: event.currentTarget.value })} />
    px
  </Item>
</Group>

export default Size
