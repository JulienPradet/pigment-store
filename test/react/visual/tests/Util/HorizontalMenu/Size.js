import React from 'react'
import PigmentStore from 'pigment-store'
import Size from '../../../../../../src/core/display/App/util/View/HorizontalMenu/Size'

const aSizeConfig = {
  size: {
    width: '1920',
    height: 'auto'
  },
  onChange: function onChange (newSize) {
    console.log(newSize)
  }
}

export default PigmentStore.React.describe('Size', Size)
  .featureJsx(
    'Default',
    <Size {...aSizeConfig} />
  )
  .featureJsx(
    'With auto values',
    <Size {...aSizeConfig} size={{height: 'auto', width: 'auto'}} />
  )
  .featureJsx(
    'With fixed values',
    <Size {...aSizeConfig} size={{height: '720', width: '360'}} />
  )
