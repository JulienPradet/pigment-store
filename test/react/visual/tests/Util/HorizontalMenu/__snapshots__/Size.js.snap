exports[`Size Default 1`] = `
<div
  className="size">
  <div
    className="sizeInput">
    <div>
      <span
        checked={true}
        className="rc-switch rc-switch-checked"
        defaultChecked={false}
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        tabIndex="0">
        <span
          className="rc-switch-inner" />
      </span>
    </div>
    <div>
      Width
    </div>
    <div>
      <input
        disabled={false}
        onChange={[Function]}
        type="number"
        value="1920" />
      px
    </div>
  </div>
  <div
    className="sizeInput">
    <div>
      <span
        checked={false}
        className="rc-switch"
        defaultChecked={false}
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        tabIndex="0">
        <span
          className="rc-switch-inner" />
      </span>
    </div>
    <div>
      Height
    </div>
    <div>
      <input
        disabled={true}
        onChange={[Function]}
        type="number"
        value={0} />
      px
    </div>
  </div>
</div>
`;

exports[`Size With auto values 1`] = `
<div
  className="size">
  <div
    className="sizeInput">
    <div>
      <span
        checked={false}
        className="rc-switch"
        defaultChecked={false}
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        tabIndex="0">
        <span
          className="rc-switch-inner" />
      </span>
    </div>
    <div>
      Width
    </div>
    <div>
      <input
        disabled={true}
        onChange={[Function]}
        type="number"
        value={0} />
      px
    </div>
  </div>
  <div
    className="sizeInput">
    <div>
      <span
        checked={false}
        className="rc-switch"
        defaultChecked={false}
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        tabIndex="0">
        <span
          className="rc-switch-inner" />
      </span>
    </div>
    <div>
      Height
    </div>
    <div>
      <input
        disabled={true}
        onChange={[Function]}
        type="number"
        value={0} />
      px
    </div>
  </div>
</div>
`;

exports[`Size With fixed values 1`] = `
<div
  className="size">
  <div
    className="sizeInput">
    <div>
      <span
        checked={true}
        className="rc-switch rc-switch-checked"
        defaultChecked={false}
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        tabIndex="0">
        <span
          className="rc-switch-inner" />
      </span>
    </div>
    <div>
      Width
    </div>
    <div>
      <input
        disabled={false}
        onChange={[Function]}
        type="number"
        value="360" />
      px
    </div>
  </div>
  <div
    className="sizeInput">
    <div>
      <span
        checked={true}
        className="rc-switch rc-switch-checked"
        defaultChecked={false}
        onChange={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseUp={[Function]}
        tabIndex="0">
        <span
          className="rc-switch-inner" />
      </span>
    </div>
    <div>
      Height
    </div>
    <div>
      <input
        disabled={false}
        onChange={[Function]}
        type="number"
        value="720" />
      px
    </div>
  </div>
</div>
`;
